src/lib/common/NotificationFormat.cpp

```diff
@@ -0,0 +1,76 @@
+/*
+*
+* Copyright 2016 Telefonica Investigacion y Desarrollo, S.A.U
+*
+* This file is part of Orion Context Broker.
+*
+* Orion Context Broker is free software: you can redistribute it and/or
+* modify it under the terms of the GNU Affero General Public License as
+* published by the Free Software Foundation, either version 3 of the
+* License, or (at your option) any later version.
+*
+* Orion Context Broker is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
+* General Public License for more details.
+*
+* You should have received a copy of the GNU Affero General Public License
+* along with Orion Context Broker. If not, see http://www.gnu.org/licenses/.
+*
+* For those usages not covered by this license please contact with
+* iot_support at tid dot es
+*
+* Author: Ken Zangelin
+*/
+#include <string.h>
+#include <string>
+#include <sstream>
+
+#include "logMsg/logMsg.h"
+#include "logMsg/traceLevels.h"
+#include "common/NotificationFormat.h"
+#include "common/wsStrip.h"
+
+
+
+/* ****************************************************************************
+*
+* notificationFormatToString - 
+*/
+const char* notificationFormatToString(NotificationFormat format, bool noDefault)
```

Comment:
So, I did the compilation test and got the following errors:

    common/NotificationFormat.cpp: In function 'const std::string& notificationFormatToString(NotificationFormat, bool)':
    common/NotificationFormat.cpp:44: error: returning reference to temporary
    common/NotificationFormat.cpp:45: error: returning reference to temporary
    common/NotificationFormat.cpp:46: error: returning reference to temporary
    common/NotificationFormat.cpp:47: error: returning reference to temporary

Lines 44-47:

    case NGSI_V1_JSON:                return "JSON";
    case NGSI_V2_NORMALIZED:          return "normalized";
    case NGSI_V2_KEYVALUES:           return "keyValues";
    case NGSI_V2_VALUES:              return "values";

Lesson learned, you can't return a 'read-only' string using `const std::string&` as return type.

This could of course be avoided, by creating std::strings for each constant string that you want to return, like this:

    const std::string json       = "JSON";
    const std::string normalized = "normalized";
    const std::string keyValues  = "keyValues";
    const std::string values     = "values";
    
    switch (format)
    {
    case NGSI_V1_JSON:          return json;
    case NGSI_V2_NORMALIZED:    return normalized;
    case NGSI_V2_KEYVALUES:     return keyValues;
    case NGSI_V2_VALUES:        return values;
    ...

However, there must be a limit as to how much we have to sacrifice only to use std::string instead of char*.
If what we want here is *symmetry* between stringToNotificationFormat and notificationFormatToString I would propose to change the `const std::string` in stringToNotificationFormat for a `const char*`.
However, that would go against our 'general' rule to use `const std::string&` for input strings and `char*` for output strings ...

Perhaps it is best to just leave this **as is**.